lib/pure/concurrency/threadpool.nim

```diff
@@ -273,6 +273,31 @@ proc blockUntilAny*(flowVars: openArray[FlowVarBase]): int =
   ## to be able to wait on, -1 is returned.
   ##
   ## **Note:** This results in non-deterministic behaviour and should be avoided.
+  runnableExamples("--threads:on"):
+    from std/os import sleep
+
+    proc worker(n: int): int =
+      result = (100*n mod 29) * 7
+      sleep(result) # Pseudo-random: 91, 182, 70, 161, 49, 140, 28, 119, 7, 98
+
+    const N = 10
+    var flowvars = newSeq[FlowVarBase](N)
```

Comment:
Why is this not `FlowVar[int]`, do not need to convert later on if this is.

Reply:
Because
```
proc blockUntilAny(flowVars: openArray[FlowVarBase]): int
  first type mismatch at position: 1
  required type for flowVars: openArray[FlowVarBase]
  but expression 'flowvars' is of type: seq[FlowVar[system.int]]
```

Reply:
Shouldn't we change the signature of the proc then?
```proc blockUntilAny*[T: FlowVarBase](flowVars: openArray[T]): int =```